tfquaternion/tfquaternion.py

```diff
@@ -180,34 +236,34 @@ class Quaternion(object):
     # https://stackoverflow.com/questions/40694380/forcing-multiplication-to-use-rmul-instead-of-numpy-array-mul-or-byp)
     __array_priority__ = 1000
 
-    def __init__(self, wxyz=(1, 0, 0, 0), dtype=tf.float32, name=None):
+    def __init__(self, wxyz=(1, 0, 0, 0), dtype=None, name=None):
         """The quaternion constructor.
-
         Args:
             wxyz: The values for w, x, y, z, a `tf.Tensor` with shape (..., 4).
                 Note that quaternions only support floating point numbers.
                 Defaults to (1.0, 0.0, 0.0, 0.0)
             dtype: The type used for the quaternion, must be a floating point
                 number, i.e. one of tf.float16, tf.float32, tf.float64.
             name: An optional name for the tensor.
-
         Returns:
             A Quaternion.
-
         Raises:
             ValueError, if wxyz is a `tf.Tensor` and the tensors dtype differs
                 from the given dtype.
             ValueError, if the last dimension of wxyz is not 4.
             TypeError, if dtype is not a float.
         """
+        try:
```

Comment:
How about making this explicit instead of the try/except? I.e.,
```
if dtype is None:
   dtype = wxyz.dtype if isinstance(wxyz, (tf.Tensor, tf.Variable. np.ndarray)) else tf.float32
```